File: INICIANTE/1040/main.dart
import 'dart:io';

void main() {
  var [n1, n2, n3, n4] = stdin.readLineSync()!.split(' ').map(double.parse).toList();
  double finalGrade = (n1 * 2 + n2 * 3 + n3 * 4 + n4) / 10;
  print('Media: ${finalGrade.toStringAsFixed(1)}');
  if (finalGrade < 5.0)
    print('Aluno reprovado.');
  else if (finalGrade > 7.0)
    print('Aluno aprovado.');
  else {
    print('Aluno em exame.');
    double retakeGrade = double.parse(stdin.readLineSync()!);
    print('Nota do exame: ${retakeGrade}');
    finalGrade = (finalGrade + retakeGrade) / 2;
    print(finalGrade < 5.0 ? 'Aluno reprovado.' : 'Aluno aprovado.');
    print('Media final: ${finalGrade.toStringAsFixed(1)}');
  }
}

File: INICIANTE/1040/main.c
#include <stdio.h>

int main() {
    float n1, n2, n3, n4;
    scanf("%f %f %f %f", &n1, &n2, &n3, &n4);
    float finalGrade = (n1 * 2 + n2 * 3 + n3 * 4 + n4) / 10;
    printf("Media: %.1f\n", finalGrade);
    if (finalGrade < 5.0)
        printf("Aluno reprovado.\n");
    else if (finalGrade > 7.0)
        printf("Aluno aprovado.\n");
    else {
        printf("Aluno em exame.\n");
        float retakeGrade;
        scanf("%f", &retakeGrade);
        printf("Nota do exame: %.1f\n", retakeGrade);
        finalGrade = (finalGrade + retakeGrade) / 2;
        printf(finalGrade < 5.0 ? "Aluno reprovado.\n" : "Aluno aprovado.\n");
        printf("Media final: %.1f\n", finalGrade);
    }
    return 0;
}
